### PR TITLE
IP-429 - Add otherFile to CancerInterpretationRequest

### PR DESCRIPTION
**Pull request by greglever**
[IP-429 - Add otherFile to CancerInterpretationRequest](https://jira.extge.co.uk/browse/IP-429)
======================

Summary of changes
===================
* Add `otherFiles` to `InterpretationRequestCancer` versions 4.0.0 and 4.2.0
* Extend `FileType` enum as per spec in ticket

- [x] Building locally
- [x] Tests passing locally
```
 ---> 8dde4805867f
Removing intermediate container 3948f4d597e7
Successfully built 8dde4805867f
Successfully tagged gel:latest
```
```
(.env) root@a3dfbe24fcb5:/gel/GelReportModels# python -m unittest discover          
...............................
----------------------------------------------------------------------
Ran 31 tests in 2.278s

OK
(.env) root@a3dfbe24fcb5:/gel/GelReportModels# 

```

### CODE REVIEW

**Comment by priesgo** [comment]
Can we add documentation for the new FileTypes in the comment above the record? That is important as `avrodocs` uses the comments to generate documentation.
Also, unless this is fixed for any external reason it will be better to maintain the "all bigcase" format which is consistent across almost all file types.

File: schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl [hunk 1]
```diff
@@ -39,7 +39,10 @@ Types:
         BigWig,
         MD5Sum,
         ROH,
-        OTHER
+        OTHER,
+        partition,
+        VFResults,
+        coverage
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl]: @cjohnson12 are these values fixed in this case ?
Is there any documentation for what they are and why they're needed ?

Comment (cjohnson12) [reply schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl]: All upper case please.
I will get an answer from Illumina on these at the call at 2pm

File: schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl [hunk 2]
```diff
@@ -39,7 +42,10 @@ Types:
         BigWig,
         MD5Sum,
         ROH,
-        OTHER
+        OTHER,
+        PARTITION,
+        VARIANT_FREQUENCIES,
+        COVERAGE
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl]: @priesgo are you happy with the uppercase ?

File: schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl [hunk 3]
```diff
@@ -23,6 +23,9 @@ Types:
 * `MD5Sum`: a MD5Sum file
 * `ROH`: a BED file with Regions of homozygosity
 * `OTHER`: other unspecified file type
+* `PARTITION`: TODO
+* `VARIANT_FREQUENCIES`: TODO
+* `COVERAGE`: TODO
```
Comment (greglever) [inline schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl]: @cjohnson12 will get clarification on these from Illumina

File: schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl (hunk 2 above)
Comment (priesgo) [reply schemas/IDLs/org.gel.models.report.avro/4.0.0/CommonRequest.avdl]: I am

**Comment by cjohnson12** [comment]
VFResults - used for b-allele frequency visualisation
Partitioned -Comes from Canvas and gives coverage in high density areas, used for visualisation
Coverage.json - coverage for whole genome from GeL analysis

**Pull request merged**